File: 0125-valid-palindrome/0125-valid-palindrome.cpp
class Solution {
public:
    bool isPalindrome(string s) {
        string filtered = "";

        for(int i =0;i<s.length();i++)
        {
            if(isalnum(s[i]))
            {
                filtered += tolower(s[i]);
            }
        }

        int i =0;
        int j = filtered.length()-1;

        while(j>i)
        {
            if(filtered[i] != filtered[j])
            {
                return false;
            }
            i++;
            j--;
        }

        return true;
    }
};

File: 0125-valid-palindrome/0125-valid-palindrome.py
class Solution:
    def isPalindrome(self, s: str) -> bool:
        s = s.lower()


        l = 0
        r= len(s)-1

        while l<r:
            while s[l] == " ":
                l += 1

            while s[r] == " ":
                r -= 1

            while not s[l].isalnum():
                l += 1

                if l >= len(s):
                    return True

            while not s[r].isalnum():
                r -= 1

                if r <= 0:
                    return True

            if l < len(s) and r > 0:
                if s[l] != s[r]:
                    return False

            l += 1
            r -= 1

        return True                
